game-state-service

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import json
+import traceback
+
+from gi.repository import Gio
+from gi.repository import GLib
+
+
+class GameState(object):
+
+    _DEFERRED_TIME = 250
+
+    def __init__(self):
+        self._state = {}
+        self._state_save_deferred_id = None
+        self._state_from_file()
+
+    def _state_get_file(self):
+        path = GLib.build_filenamev([GLib.get_user_data_dir(), 'state.json'])
+        return Gio.File.new_for_path(path)
+
+    def _state_from_file(self):
+        state_file = self._state_get_file()
+        try:
+            success, state_data, tag = state_file.load_contents(None)
+            self._state = json.loads(state_data.decode())
+        except (GLib.Error, json.decoder.JSONDecodeError):
+            traceback.print_exc()
+
+    def _state_to_file(self):
+        state_data = json.dumps(self._state,
+                                sort_keys=True,
+                                indent=2,
+                                separators=(',', ': '))
+        state_file = self._state_get_file()
+        try:
+            state_file.replace_contents(
+                bytes(state_data.encode()),
+                None,
+                False,
+                0,
+                None)
+        except GLib.Error:
+            traceback.print_exc()
+        finally:
+            self._state_save_deferred_id = None
```

Comment:
This is also done in the only caller of `_state_to_file()`, so it's redundant.

I think it would be better to remove this one and keep the other one, so that the reset of `_state_save_deferred_id` is next to `SOURCE_REMOVE`.